### ISSUE TITLE
Add model history to GD models

### ISSUE THREAD

**Issue by TheCSGuy25** [TheCSGuy25/CommML#5]
Add Training Accuracy, Training Loss history to models that undergo gradient descent while training.
